src/compile.rs

```diff
@@ -1592,40 +1597,28 @@ fn create_synthetic_resource(
         parent_names.append(&mut inherit_ann.clone());
     }
 
-    dup_res_decl.inherits = parent_names;
+    let mut new_decl = TypeDecl::new(res_name.clone(), parent_names, Vec::new());
     // Virtual resources become concrete when associated to concrete types
-    dup_res_decl.is_virtual = dup_res_decl.is_virtual && dom_info.is_virtual;
-    let dup_res_is_virtual = dup_res_decl.is_virtual;
+    new_decl.is_virtual = classes.iter().all(|c| c.is_virtual) && dom_info.is_virtual;
+    new_decl.is_trait = false;
+    new_decl.is_extension = false;
     // The synthetic resource keeps some, but not all annotations from its parent.
     // Specifically, Makelist and derive are kept from the parent
     // TODO: This would be cleaner if we convert to AnnotationInfos first and implent the logic as
     // a member funtion in AnnotationInfo
     // See https://github.com/dburgener/cascade/pull/39#discussion_r999510493 for fuller discussion
-    dup_res_decl
-        .annotations
-        .annotations
-        .retain(|a| a.name.as_ref() == "makelist" || a.name.as_ref() == "derive");
-
-    dup_res_decl
-        .expressions
-        .iter_mut()
-        .for_each(|e| e.set_class_name_if_decl(res_name.clone()));
-
-    dup_res_decl
-        .expressions
-        // If dup_res_decl is concrete, do not inherit virtual functions
-        // Never inherit statements
-        .retain(|e| {
-            (dup_res_is_virtual || !e.is_virtual_function())
-                && matches!(e, Expression::Decl(Declaration::Func(_)))
-        });
-    if !global_exprs.insert(Expression::Decl(Declaration::Type(Box::new(
-        dup_res_decl.clone(),
-    )))) {
+    new_decl.annotations.annotations = classes
+        .iter()
+        .flat_map(|c| c.decl.iter().flat_map(|d| d.annotations.annotations.iter()))
+        .filter(|a| a.name.as_ref() == "makelist" || a.name.as_ref() == "derive")
+        .cloned()
+        .collect(); // TODO: dedup?
```

Comment:
Looks like we don't need the de-dup?

Reply:
Well, at a minimum we pass the existing tests without deduping.  The test coverage with annotations is pretty poor though.  I'm thinking that an issue for later follow-up to both extend test coverage and look into this makes sense.

I do think that duplicate `@makelist`s are just wasting storage.  I suppose duplicate `@derive`s might cause it to derive the same function twice, which would be a problem.